incubator/hnc/pkg/kubectl/set.go

```diff
@@ -16,100 +16,153 @@ limitations under the License.
 package kubectl
 
 import (
+	"errors"
 	"fmt"
 	"os"
 
 	"github.com/spf13/cobra"
+
+	api "github.com/kubernetes-sigs/multi-tenancy/incubator/hnc/api/v1alpha1"
 )
 
+type flagStatus struct {
+	root                bool
+	parent              bool
+	parentString        string
+	requiredChild       bool
+	requiredChildArray  []string
+	optionalChild       bool
+	optionalChildString string
+}
+
 var setCmd = &cobra.Command{
 	Use:     "set",
 	Short:   "Allows setting namespace hierarchy",
 	Aliases: []string{"set"},
 	Args:    cobra.MinimumNArgs(1),
 	Run: func(cmd *cobra.Command, args []string) {
 		nnm := args[0]
-		hc := client.getHierarchy(nnm)
-		oldpnm := hc.Spec.Parent
 		flags := cmd.Flags()
-		numChanges := 0
+		parentString, _ := flags.GetString("parent")
+		requiredChildArray, _ := flags.GetStringArray("requiredChild")
+		optionalChildString, _ := flags.GetString("optionalChild")
+
+		flagStatus := flagStatus{
+			root:                flags.Changed("root"),
+			parent:              flags.Changed("parent"),
+			requiredChild:       flags.Changed("requiredChild"),
+			optionalChild:       flags.Changed("optionalChild"),
+			parentString:        parentString,
+			requiredChildArray:  requiredChildArray,
+			optionalChildString: optionalChildString,
+		}
 
-		if flags.Changed("root") && flags.Changed("parent") {
-			fmt.Println("Cannot set both --root and --parent at the same time")
+		msgs, err := setCmdFunc(client, flagStatus, nnm)
+		if err != nil {
+			fmt.Printf("%+v", err)
```

Comment:
We should just print the string. If err is any kind of complex struct, we don't want to show all the internal details. So this should just be:

`fmt.Println(err.Error())`

Reply:
ok